sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/HiveThriftServer2Listener.scala

```diff
@@ -0,0 +1,414 @@
+      case e: SparkListenerStatementStart => processEventStatementStart(e)
+      case e: SparkListenerStatementParsed => processEventStatementParsed(e)
+      case e: SparkListenerStatementCanceled => processEventStatementCanceled(e)
+      case e: SparkListenerStatementError => processEventStatementError(e)
+      case e: SparkListenerStatementFinish => processEventStatementFinish(e)
+      case e: SparkListenerOperationClosed => processEventOperationClosed(e)
+      case _ => // Ignore
+    }
+  }
+
+  private def processEventSessionCreated(e: SparkListenerSessionCreated): Unit = {
+    val session = getOrCreateSession(e.sessionId, e.startTime, e.ip, e.userName)
+    sessionList.put(e.sessionId, session)
+    updateLiveStore(session, true)
+  }
+
+  private def processEventSessionClosed(e: SparkListenerSessionClosed): Unit = {
+    val session = sessionList.get(e.sessionId)
+    session.finishTimestamp = e.finishTime
+    updateLiveStore(session, true)
+    sessionList.remove(e.sessionId)
+
+  }
+
+  private def processEventStatementStart(e: SparkListenerStatementStart): Unit = {
+    val info = getOrCreateExecution(
+      e.id,
+      e.statement,
+      e.sessionId,
+      e.startTime,
+      e.userName)
+
+    info.state = ExecutionState.STARTED
+    executionList.put(e.id, info)
+    sessionList.get(e.sessionId).totalExecution += 1
+    executionList.get(e.id).groupId = e.groupId
+    updateLiveStore(executionList.get(e.id))
+    updateLiveStore(sessionList.get(e.sessionId), true)
+  }
+
+  private def processEventStatementParsed(e: SparkListenerStatementParsed): Unit = {
+    executionList.get(e.id).executePlan = e.executionPlan
+    executionList.get(e.id).state = ExecutionState.COMPILED
+    updateLiveStore(executionList.get(e.id))
+  }
+
+  private def processEventStatementCanceled(e: SparkListenerStatementCanceled): Unit = {
+    executionList.get(e.id).finishTimestamp = e.finishTime
+    executionList.get(e.id).state = ExecutionState.CANCELED
+    updateLiveStore(executionList.get(e.id))
+  }
+
+  private def processEventStatementError(e: SparkListenerStatementError): Unit = {
+    executionList.get(e.id).finishTimestamp = e.finishTime
+    executionList.get(e.id).detail = e.errorMsg
+    executionList.get(e.id).state = ExecutionState.FAILED
+    updateLiveStore(executionList.get(e.id))
+  }
+
+  private def processEventStatementFinish(e: SparkListenerStatementFinish): Unit = {
+    executionList.get(e.id).finishTimestamp = e.finishTime
+    executionList.get(e.id).state = ExecutionState.FINISHED
+    updateLiveStore(executionList.get(e.id))
+  }
+
+  private def processEventOperationClosed(e: SparkListenerOperationClosed): Unit = {
+    executionList.get(e.id).closeTimestamp = e.closeTime
+    executionList.get(e.id).state = ExecutionState.CLOSED
+    updateLiveStore(executionList.get(e.id), true)
+    executionList.remove(e.id)
+  }
+
+
+  def onSessionCreated(ip: String, sessionId: String, userName: String = "UNKNOWN"): Unit = {
```

Comment:
Hmm... After the previous discussion, I would prefer to just change all the Operations to create these events and post directly to the sc listener bus, instead of adding these helpers here. AppStatusListener and SQLAppStatusListener don't have such helpers.
@AngersZhuuuu WDYT? Does it work for you when it's added to the "status" listener queue, so that the likelihood of dropping events is minimized? As a followup there could be introduced a "criticial events" queue that doesn't have limited capacity (or blocks instead of dropping) to make sure that no events are dropped.

Reply:
Thanks @juliuszsompolski for the review.
Now I have created a new class `HiveThriftServer2EventManager`, which creates listener events and post to listener bus. Now `HiveThriftServer2Listener` class only listens the events. 

Also I moved all the listener/Store classes to the `ui` directory, which also follows similar to AppStatusListener and SQLAppStatusListner

Reply:
> @AngersZhuuuu WDYT? Does it work for you when it's added to the "status" listener queue, so that the likelihood of dropping events is minimized? As a followup there could be introduced a "criticial events" queue that doesn't have limited capacity (or blocks instead of dropping) to make sure that no events are dropped.

Adding it to status queue is OK for me. Thanks